lib/geomancer.ex

```diff
@@ -1,21 +1,19 @@
 defmodule Geomancer do
   @moduledoc """
-  A library to convert between various geospatial file formats
+  A behaviour to convert geospatial data from one format to another.
   """
 
-  alias Geomancer.Shapefile
+  @type input_path() :: String.t()
+  @type conversion() :: String.t()
+  @type reason() :: String.t()
+  @type geo_struct() :: struct | map
 
-  @type geo_json :: String.t()
-  @type reason :: String.t()
+  @callback from(input_path) :: {:ok, conversion} | {:error, reason}
+  @callback read(input_path) :: {:ok, geo_struct} | {:error, reason}
```

Comment:
I find `from` and `read` to be synonymous in this context. Could you perhaps rename `from` to something like `transform` or `translate`?

Reply:
Also, is the idea that this library will be handling more file formats in the future? If not, I'm skeptical of using behaviours here. (But I think it is)

Reply:
They're now `read/1` and `convert/1`. Also, I do expect to have more file formats in the future.